### PR TITLE
Update ballerina/auth module for Swan Lake Design

### PR DESCRIPTION
**Pull request by ldclakmal**
## Purpose
This PR updates the module for the proposed auth design for Ballerina Swan Lake [1]. The current `ballerina/ldap` module is considered as a part of the `ballerina/auth` module. Therefore the content of `ballerina/ldap` is moved here with the Git history.
[1] https://docs.google.com/document/d/1dGw5uUP6kqZNTwMfQ_Ik-k0HTMKhX70XpEA3tys9_kk/edit?usp=sharing

Fixes https://github.com/ballerina-platform/ballerina-standard-library/issues/715
Fixes https://github.com/ballerina-platform/ballerina-standard-library/issues/718

### CODE REVIEW

File: auth-ballerina/Ballerina.toml [hunk 1]
```diff
@@ -1,35 +1,35 @@
 [package]
 org = "ballerina"
 name = "auth"
-version = "@toml.version@"
+version = "1.0.5"
```
Comment (Maninda) [inline auth-ballerina/Ballerina.toml:4]: Seems to be mistakenly committed removing the annotations.

Comment (ldclakmal) [reply auth-ballerina/Ballerina.toml]: Thanks. Fixed with https://github.com/ballerina-platform/module-ballerina-auth/pull/50/commits/371a25667a87400fcb53a6eb35aef171ee56ec8b

File: issue_template.md [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
-
-**Affected Product Version:**
-
-**OS, DB, other environment details and versions:**    
-
-**Steps to reproduce:**
-
-
-**Related Issues:**
+**Description:**
```
Comment (ThisaruGuruge) [inline issue_template.md:1]: Do we need this? We don't create issues in this repo, right?

Comment (ldclakmal) [reply issue_template.md]: Yes. Removed with https://github.com/ballerina-platform/module-ballerina-auth/pull/50/commits/34d703c2f358ed32738c5836ec77e57a0fc5528c

**Review by chamil321 — APPROVED** [review]

**Pull request merged**